### PR TITLE
Update androidx-compose-bom dependency to use '2024.09.01'

### PR DESCRIPTION
**Pull request by hafizrahman**
Please do not merge before target is `trunk`. It is currently targeting a different branch that is also about dependency update #13046. It seems safest to get that dependency update merged first, and then this one after, to avoid any trouble.

### Description

This PR updates the version used on `androidx-compose-bom` to '2024.09.00'.

**Why '2024.09.00'?** Originally, I wanted to add these dependencies:
```
androidx.compose.material3.adaptive:adaptive | 1.0.0
androidx.compose.material3.adaptive:adaptive-layout | 1.0.0
androidx.compose.material3.adaptive:adaptive-navigation | 1.0.0
```

and these were made available since '2024.09.00'. To confirm [you can go here](https://developer.android.com/develop/ui/compose/bom/bom-mapping) and change the version in the dropdown to `2024.09.00` and compare it to the version before it.

Update: this is now updated to **2024.09.01** instead. The reason is [from the previous PR:](https://github.com/woocommerce/woocommerce-android/pull/13046#issuecomment-2519971988):

> Lifecycle Runtime Compose now depends on [Compose Runtime 1.7.1](https://developer.android.com/jetpack/androidx/releases/compose-runtime#1.7.1)

I had thought about updating to a newer version, but as you can see from the commits, this alone already introduces quite a bit of required changes, so I'm a bit wary about going even further.

I apologize for the size of the PR. Most of them are repeat changes related to these:

| No. | Updates | Notes |
|-|-|-|
| 1. | deprecated `rememberRipple()` usage | Trivial syntax change |
| 2. | deprecated `RippleTheme` and `LocalRippleTheme` usage. | More info in Testing Information section |
| 3. | deprecated `LocalLifecycleOwner` usage. | Trivial syntax change |
| 4. | deprecated `animateItemPlacement()` usage. | Trivial syntax change |
| 5. | deprecated `KeyboardOptions` constructor. | Trivial syntax change |
| 6. | deprecated `updateTransition` usage. | Trivial syntax change |
| 7. |  deprecated `ClickableText` usage. | More info in Testing Information section |
| 8. | deprecated `LocalTextInputService` usage | More info in Testing Information section |

These are all warnings-level change, not error. But since build.gradle is set to treat `allWarningsAsErrors`, they need to be updated.

### Steps to reproduce

- Make sure the app builds with no errors
- Smoke test the main flows P91TBi-bVe-p2

### Testing information

**On ClickableText:**
I've tested them and can confirm that they are working fine, but could use a double-check.  Please also double-check the code changes on commit 5cc7b7a1251a16cde1c6f07ad3f48227ada98196 and da33b40c78441c3ee9287e67f469577440c580ca to emsure I'm not copy/pasting the wrong string resource/onclick functions.

The affected screens are:
- JetpackConsent
- AddProductWithAIBottomSheet
- BlazeCampaignPaymentSummaryScreen
- WidgetError
- AccountMismatchErrorScreen
- PaymentsPreSetupScreen
- WooPaymentsSetupInstructionsScreen
- DomainDashboardScreen

**On RippleTheme:**
The `RippleTheme` replacement (in this commit https://github.com/woocommerce/woocommerce-android/pull/13060/commits/145a2170e2277629b9d4736a3233389b4810672c) now does not have a custom Alpha compared to the previous one. This is because now [RippleAlpha](https://developer.android.com/reference/kotlin/androidx/compose/material/ripple/RippleAlpha) has different constructor while `defaultRippleAlpha` is deprecated, and I can't figure out how to replicate it. With an eye test I don't see a difference between the ripple effect in `trunk` and this branch (I test using the "Share" button in Product Detail's share bottom sheet), but suggestions welcome if updating it further is needed.

**On the focus fix** (commits a3716c6bd2bdd5b26204ec284b569d7c2431ce9e and efcef5c20177e6cf22e7b7efaa7f1494bb0753ef)
This is related to Aztec usage in Custom Fields Editor and has been discussed in Slack p1733297997528939-slack-C03L1NF1EA3 . @hichamboushaba and I have tested this fix so it's okay to not be tested again. The **tl;dr** is that the original code contained a deprecated use of `LocalTextInputService`, but we're lucky that it's addressing an issue [that's already fixed](https://issuetracker.google.com/issues/318530776), so we could clean up the code and avoid the deprecation warning at the same time. 

### The tests that have been performed

I've smoke tested this on the flow P91TBi-bVe-p2
Also checked the affected clickable texts, and RippleTheme effect, and the Aztec keyboard related fix above.

### Images/gif

n/a

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@
 androidx-browser = '1.5.0'
 androidx-camera = '1.2.3'
 androidx-cardview = '1.0.0'
-androidx-compose-bom = '2024.04.00'
+androidx-compose-bom = '2024.09.00'
```
[the remark this answers is not in the document]
Comment (hafizrahman) [reply gradle/libs.versions.toml]: I have explained in the original post above why I chose 2024.09.00

**Comment by hafizrahman** [comment]
@JorgeMucientes as mentioned in https://github.com/woocommerce/woocommerce-android/pull/13046#issuecomment-2522173308 , I'll bump this to 2024.09.01 instead of 2024.09.00, so I'm setting this back to draft for now. Will let you know once it's done.

**Comment by hafizrahman** [comment]
@JorgeMucientes Not seeing any build issue after upgrading to 2024.09.01, so setting this to ready for review again 🙏🏼

**Review by JorgeMucientes — APPROVED** [review]
Great job @hafizrahman. Thanks for proactively tackling this update. 

Code changes look good and I smoke tested the app and everything worked well. 

About the Ripple effect
> With an eye test I don't see a difference between the ripple effect in trunk and this branch (I test using the "Share" button in Product Detail's share bottom sheet), but suggestions welcome if updating it further is needed.

I think the difference is noticeable only in dark mode, where the ripple effect is now applying `<color name="color_primary_variant">@color/woo_purple_50</color>` which is a darker color than the `color_primary` set. I'm not sure how it worked before, but visually you could see the button got brighter upon tapping on it. Now it gets slightly darker. Barely noticeable. This is a minor thing, so definitely not blocking and can be addressed in a different PR. Maybe we can refine the color used in dark mode to have a darker ripple similar to the one the FAB button has.  

PS: I tested updating the BOM to the latest version during my hackweek (using this brach as base) and I didn't have to apply any code changes to get the app to compile. So hopefully the next BOM update will be painless, if we don't pospone updating to much :)

**Comment by hafizrahman** [comment]
@JorgeMucientes Thanks for checking out the Ripple situation. We're also likely updating colors soon, so probably a good idea to not spend too much time on this now.

**Pull request merged**